### PR TITLE
Adding support for multiple callbacks per client side method

### PR DESCRIPTION
**Pull request by moozzyk**
...and a possibility to remove callbacks

Fixes: #807

### CODE REVIEW

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts [hunk 1]
```diff
@@ -211,7 +211,32 @@ export class HubConnection {
...
+        if (!methodName || !method) {
+            return;
+        }
+
+        methodName = methodName.toLowerCase();
+        let handlers = this.methods.get(methodName);
+        if (!handlers) {
+            return;
+        }
+        var removeIdx = handlers.indexOf(method);
```
Comment (davidfowl) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts]: This works?

Comment (moozzyk) [reply client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts]: Yes - as long as this is the same function. I also added a test.

Comment (davidfowl) [reply client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts]: Heh cool. I guess it's just object instance.

File: client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/HubConnection.spec.ts [hunk 2]
```diff
@@ -135,6 +137,147 @@ describe("HubConnection", () => {
...
 
+    describe("on", () => {
+        it("invocations ignored in callbacks not registered", async () => {
+            let warnings: string[] = [];
+            let logger = <ILogger>{
+                log: function(logLevel: LogLevel, message: string) {
+                    if (logLevel === LogLevel.Warning) {
+                        warnings.push(message);
+                    }
+
```
Comment (mikaelm12) [inline client-ts/Microsoft.AspNetCore.SignalR.Client.TS.Tests/HubConnection.spec.ts]: nit: Remove new line

**Pull request merged**